platform/autoupdate.services/src/org/netbeans/modules/autoupdate/services/FeatureUpdateElementImpl.java

```diff
@@ -496,7 +496,7 @@ private void initializeAgent() {
                     }
                     Module m = Utilities.toModule(cnb, null);
                     if (m != null && ! m.getProblems().isEmpty()) {
-                        dependenciesToModulesOrFeatures.remove(depSpec);
+                        it.remove();
```

Comment:
I ran against the same wall yesterday and came to the same conclusion. As this is a double nested loop, I think it would be good to either terminate the inner loop with a break or the continue the outer loop with a label. Else we could hit the remove a second time and enter undefined territory.

Reply:
You're right, we need to break here. Calling remove for the second time would throw `IllegalStateException`.

Reply:
break added.